src/dagmc/dagmcmetadata.cpp

```diff
@@ -0,0 +1,591 @@
+  // if this is an index query
+  moab::EntityHandle eh;
+  if (idx == true) {
+    eh = DAG->entity_by_index(3, vol);
+  } else {
+    eh = DAG->entity_by_id(3, vol);
+  }
+  return get_volume_property(property, eh);
+}
+
+// Get a given property on a surface
+std::string dagmcMetaData::get_surface_property(std::string property, moab::EntityHandle eh) {
+  std::string value = "";
+  if (property == "boundary") {
+    value = surface_boundary_data_eh[eh];
+  } else if (property == "tally") {
+    value = tally_data_eh[eh];
+  } else {
+    std::cout << "Not a valid property for surfaces" << std::endl;
+  }
+  return value;
+}
+
+// overloaded get_surface_property for indices and id's'
+std::string dagmcMetaData::get_surface_property(std::string property, int vol, bool idx) {
+  // if this is an index query
+  moab::EntityHandle eh;
+  if (idx == true) {
+    eh = DAG->entity_by_index(2, vol);
+  } else {
+    eh = DAG->entity_by_id(2, vol);
+  }
+  return get_surface_property(property, eh);
+}
+
+// parse the material data
+void dagmcMetaData::parse_material_data() {
+  std::map<moab::EntityHandle, std::vector<std::string> > material_assignments;
+  material_assignments = get_property_assignments("mat", 3, ":/", true);
+  std::map<moab::EntityHandle, std::vector<std::string> > density_assignments;
+  density_assignments = get_property_assignments("rho", 3, ":", true);
+
+  int num_cells = DAG->num_entities(3);
+
+  std::vector<std::string> material_props;
+  std::vector<std::string> density_props;
+
+  std::string implicit_complement_material = "";
+  std::string implicit_complement_density = "";
+
+  // loop over all cells
+  for (int i = 1; i <= num_cells; ++i) {
+
+    int cellid = DAG->id_by_index(3, i);
+    moab::EntityHandle eh = DAG->entity_by_index(3, i);
+
+    material_props = material_assignments[eh];
+    density_props = density_assignments[eh];
+
+    // this is actually ok for a single volume, one that has the _comp tag at the end of it
+    if (material_props.size() > 1) {
+      // search the props for _comp
+      std::size_t comp_found;
+      int position;
+      for (int j = 0 ; j < material_props.size() ; j++) {
+        comp_found = material_props[j].find("_comp");
+        if (comp_found != std::string::npos) {
+          position = j;
+          break;
+        }
+      }
+      if (comp_found != std::string::npos) {
+        // success found the _comp tag for the impl_compl material
+        // set the impl_comp material for use later
+        implicit_complement_material = material_props[position].substr(0, material_props[position].size() - 5);
+        implicit_complement_density = density_props[0];
+        material_props.erase(material_props.begin() + position);
+      } else {
+        // failure a volume can only have a single material associated with it
+        std::cout << "more than one material for volume with id " << cellid << std::endl;
```

Comment:
Same comment as above re: `std::cout` vs `std::cerr`.  Recommend using `std::cerr` here so it is more easily caught by logging aggregators and shell formatters.